### ISSUE TITLE
Why spyder-notebook's autocomplete is better than spyder editor and native jupyter notebook?

### ISSUE THREAD

**Issue by Fred-Wu** [spyder-ide/spyder-notebook#235]
I am working with Spyder 4.01 on Ubuntu 18.04. 

For the curiosity, I dowloaded and tried spyder-notebook with Spyder IDE. 
I found out that the autocompletion works better in this spyder-notebook, and every `.<Tab>` shows function suggestions. This is not even the case for native jupyter notebook or jupyterlab.  

Normally, for pandas, the spyder editor doesn't show anything without Kite or very very slow. 
And for the native Jupyter notebook, `np.<Tab>` does not show anything. It also doesn't show the suggestions for the second `.<Tab>` in `df.groupby('B').<Tab>`.  However, everything works inside spyder-notebook. 

First I thought it was because of the installation of Kite, and then I disabled it. But the autocompletion still works, and it turns out Kite does not even support jupyter notebook. 

Is there any reason for this? And can autocompletion in spyder editor be further improved? 
```
import numpy as np
import pandas as pd
df = pd.DataFrame({'A': np.random.randn(5), 
                   'B': ['a', 'b', 'b', 'a', 'a']})
df.groupby('B').aggregate({'A': ['sum']})
```

**Comment by ccordoba12** [comment]
Did you activated the Greedy completer in our preferences?

![Selección_097](https://user-images.githubusercontent.com/365293/72735396-bdcaa280-3b69-11ea-85c1-b721919e5c3c.png)

That would explain what you're mentioning above.

**Comment by Fred-Wu** [comment]
Yes. I did. However, that does not work properly in IPython console too. 
For `np.<Tab>`, the suggestions appear, and when I type `r`, it will not jump to anything start with `r` to perform autocomplete, instead, the entire autocomplete window just disappears.
